tests/validation/tests/v3_api/common.py

```diff
@@ -1159,13 +1166,20 @@ def validate_response_app_endpoint(p_client, appId):
                 public_endpoint["protocol"].lower() + "://" + \
                 public_endpoint["hostname"]
             print(url)
+            start = time.time()
+            if time.time() - start > 120:
+                assert False, "failed to connect to the app"
+                return
             try:
                 r = requests.head(url)
-                assert r.status_code == 200, \
-                    "Http response is not 200. Failed to launch the app"
+                if r.status_code != 200:
+                    print("status code: ", r.status_code)
+                    time.sleep(10)
+                    validate_response_app_endpoint(p_client, appId)
```

Comment:
@sowmyav27 Can you change this to make a request to url being made in a loop instead of the method being called for waiting for status 200?